src/mlpack/core/data/string_cleaning.hpp

```diff
@@ -0,0 +1,131 @@
+
+#include <mlpack/prereqs.hpp>
+#include <boost/range/algorithm/remove_if.hpp>
+#include <boost/algorithm/string/classification.hpp>
+#include "mlpack/core/boost_backport/boost_backport_string_view.hpp"
+#include <unordered_set>
+
+namespace mlpack {
+namespace data {
+
+class StringCleaning
+{
+ public:
+  /**
+  * Function to remove punctuation from a given vector of strings.
+  *
+  * @param input Vector of strings
+  */
+  void RemovePunctuation(std::vector<std::string>& input,
+                         const std::string& punctuation = "!\"#$%&'()*+"
+                         ",-./:;<=>?@[\\]^_`{|}~")
+  {
+    for (auto& str : input)
+    {
+      str.erase(boost::remove_if(str, boost::is_any_of(punctuation)),
+          str.end());
+    }
+  }
+
+  /**
+  * Function to remove a set of chars from a given vector of strings.
+  * A function which accepts a function as an argument and removes the
+  * character if the funtion returns true.
+  * 
+  * @param input Vector of strings.
+  * @param fun A function which takes const char& c as input and returns bool;
+  *
+  * This can either be a function pointer or function object or a lamda
+  * function. Its return value should be of bool type.
+  *
+  * Definiation of function should be of type
+  * bool fn(const char& c)
+  */
+  template<typename ReturnType>
+  void RemoveChar(std::vector<std::string>& input, const ReturnType fun)
+  {
+    for (auto& str : input)
+    {
+      str.erase(boost::remove_if(str, fun), str.end());
+    }
+  }
+
+  /**
+  * Function to convert given vector of strings to lower case.
+  *
+  * @param input Vector of strings
+  */
+  void LowerCase(std::vector<std::string>& input)
+  {
+    for (auto& str : input)
+    {
+      boost::algorithm::to_lower(str);
+    }
+  }
+  /**
+  * Function to convert given vector of strings to Upper case.
+  *
+  * @param input Vector of strings
+  */
+  void UpperCase(std::vector<std::string>& input)
+  {
+    for (auto& str : input)
+    {
+      boost::algorithm::to_upper(str);
+    }
+  }
+  /**
+  * Function to remove stopwords from a given vector of strings.
+  *
+  * @param input Vector of strings
+  * @param tokenizer A function that accepts a boost::string_view as
+  *                  an argument and returns a token.
+  * This can either be a function pointer or function object or a lamda
+  * function. Its return value should be a boost::string_view, a token.
+  *
+  * Definiation of function should be of type
+  * boost::string_view fn()(boost::string_view& str)
+  */
+  template<typename TokenizerType>
+  void RemoveStopWords(std::vector<std::string>& input,
+                       std::unordered_set<boost::string_view,
+                       boost::hash<boost::string_view> >stopwords,
+                       const TokenizerType tokenizer)
+  {
+    std::string copy;
+    boost::string_view token;
+    boost::string_view strView;
+    for (auto& str : input)
+    {
+      copy.clear();
```

Comment:
Thanks @lozhnikov , I believed that there would be performance lag while creating the variable inside the loop, as in `for(){ int a}`, every iteration the variable will be created, Being a sport programmer I always believed That we should declare it at first and then use, but I researched a bit and found [this](https://stackoverflow.com/a/7959658/9559556). So i guess I will go with your suggestion.

Thanks

Reply:
Hmm.. Perhaps you were right.
I thought it doesn't influence on the performance since it doesn't increase the number of heap allocations because at the end of each iteration the string is moved to the container. However, I didn't pay attention to the allocations performed by `operator+=`. It's quite interesting I think we should make the benchmark.